go-backend/service/organization_http.go

```diff
@@ -0,0 +1,203 @@
+// @Success 200 {object}
+// @Failure 400 {object}
+func listOrganizationHandler(deps Dependencies) http.HandlerFunc {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		organizations, err := deps.Store.ListOrganizations(req.Context())
+		if err != nil {
+			logger.WithField("err", err.Error()).Error("Error fetching data")
+			rw.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		respBytes, err := json.Marshal(organizations)
+		if err != nil {
+			logger.WithField("err", err.Error()).Error("Error marshaling organizations data")
+			rw.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		rw.Header().Add("Content-Type", "application/json")
+		rw.Write(respBytes)
+	})
+}
+
+// @Title createOrganizationHandler
+// @Description Create Organizations
+// @Router /organizations [post]
+// @Accept  json
+// @Success 200 {object}
+// @Failure 400 {object}
+func createOrganizationHandler(deps Dependencies) http.HandlerFunc {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		var organization db.Organization
+		err := json.NewDecoder(req.Body).Decode(&organization)
+		if err != nil {
+			rw.WriteHeader(http.StatusInternalServerError)
+			logger.WithField("err", err.Error()).Error("Error while decoding organizations data")
+			return
+		}
+		errorResponse, valid := organization.ValidateOrganization()
+		if valid == false {
+			respBytes, err := json.Marshal(errorResponse)
+			if err != nil {
+				logger.WithField("err", err.Error()).Error("Error marshaling organizations data")
+				rw.WriteHeader(http.StatusInternalServerError)
+				return
+			}
+	
+			rw.Header().Add("Content-Type", "application/json")
+			rw.WriteHeader(http.StatusPreconditionFailed)
+			rw.Write(respBytes)
+			return
+		}
+
+		err = deps.Store.CreateOrganization(req.Context(), organization)
+		if err != nil {
+			rw.WriteHeader(http.StatusInternalServerError)
+			logger.WithField("err", err.Error()).Error("Error create organization")
+			return
+		}
+
+		rw.WriteHeader(http.StatusCreated)
+		rw.Header().Add("Content-Type", "application/json")
+	})
+}
+
+// @Title updateOrganizationHandler
+// @Description Update Organizations
+// @Router /organizations/:id [put]
+// @Accept  json
+// @Success 200 {object}
+// @Failure 400 {object}	
+func updateOrganizationHandler(deps Dependencies) http.HandlerFunc{
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		vars := mux.Vars(req)
+		id, err := strconv.Atoi(vars["id"])
+		if err != nil {
+			logger.WithField("err", err.Error()).Error("Error id key is missing")
+			rw.WriteHeader(http.StatusBadRequest)
+			return
+		}	
+		
+		var organization db.Organization
+		err = json.NewDecoder(req.Body).Decode(&organization)
+		if err != nil {
+			rw.WriteHeader(http.StatusBadRequest)
+			logger.WithField("err", err.Error()).Error("Error while decoding organization")
+			return
+		}
+
+		errorResponse, valid := organization.ValidateOrganization()
```

Comment:
we should call the method only `Validate`. 

`orgnaization.Validate` is readable in itself to convey that we are validating organization